alerts/cluster-monitoring-operator/KubeDaemonSetMisScheduled.md

```diff
@@ -0,0 +1,68 @@
+# KubeDaemonSetMisScheduled
+
+## Meaning
+
+The `KubeDaemonSetMisScheduled` alert triggers when, over a defined time interval, an excessive number of daemonset pods are found running on nodes where they should not be. This typically occurs when a pod is scheduled on a node with a taint that doesn't have a corresponding toleration, or when a node is unhealthy or overloading, hindering pod scheduling.
+
+## Impact
+
+Excessive resource usage where they could be used by other applications
+
+## Diagnosis
+
+- Check Daemonset status
+
+  ```console
+  oc -n $NAMESPACE describe daemonset $NAME
+  ```
+
+- Check the status of the pods belonging to the Daemonset and look for the fields Node, Tolerations and NodeSelector:
+
+  ```console
+  oc -n $NAMESPACE describe pod $NAME
+  ```
+
+  Replace $NAME with the name of the pod having issue.
+
+  -  Look for the `Node`, `Node-Selectors` and `Tolerations` fields in the output
+
+- Check labels for the node and verify with `NodeSelector` on Pod
+
+  ```console
+  oc get node $NAME --show-labels
+  ```
+
+  Replace $NAME with the name of the pod having issue.
+
+    - Verify whether `Node-Selector` field for pod has one of the labels of node
+
+- Check Taint for the node and verify with `Toleration` on Pod
+
+  Replace $NAME with the name of the node identified in previous step.
+
+    ```console
+    oc describe node $NAME
+    ```
+
+  - Look for the field `Taints`
+  - Verify whether `Taints `identified are tolerated (identified in `Tolerations` field in previous step)
+  
+## Mitigation
+
+- Inspect DaemonSet Configuration:
+
+  - Examine the YAML configuration of the daemonset(s) causing the issue. Ensure that the tolerations and node selectors are correctly set to match the desired nodes.
+
+- Check pod template parameters such as:
+  - pod priority - maybe it was evicted by other more important pods
+  - affinity rules - maybe due to affinities and not enough nodes it is not possible to schedule pods
+
+- Check Taints and Tolerations:
+
+  - Verify that the nodes where mis-scheduled pods are running have the expected taints.
+  - Confirm that the pods have tolerations matching those taints.
+
+- Inspect Kubernetes event logs for any relevant events that might explain why pods were scheduled where they shouldn't be.
+
+- Verify [node health](https://docs.openshift.com/container-platform/4.13/support/troubleshooting/verifying-node-health.html) 
```

Comment:
I would move this first since resource shortage is probably the main reason for the alert to fire.